nix/status-go/extract-status-go-config.nix

```diff
@@ -0,0 +1,20 @@
+#
+# This nix expression contains logic to extract the value of a field in the STATUS_GO_VERSION file, when given the field name
+#
+
+{ lib }:
+
+let
+  inherit (lib) fileContents findFirst head splitString;
+  content = fileContents ../../STATUS_GO_VERSION; # TODO: Simplify this path search with lib.locateDominatingFile
+  extractStatusGoConfig =
+    fieldName:
+      let lines = splitString "\n" content;
+          # Find the first line containing the field name
+          matchedLine = findFirst (line: (builtins.match "${fieldName}=.+" line) != null) null lines;
+      in if matchedLine != null
+         # If line is found, extract the field value
+         then head (builtins.match "${fieldName}=(.+)" matchedLine)
+         else throw "Could not find '${fieldName}' in ${content}";
+
+in extractStatusGoConfig
```

Comment:
This looks really complex, why use a custom format if you could just make it a `status-go-verison.json` and load it in Nix using `lib.importJSON`?

Reply:
Done